### PR TITLE
fix(auth) - prevent urlListener from firing auth flow twice with the same URL 

### PR DESCRIPTION
**Pull request by HilSny**
_Issue #, if available:_
#4388
_Description of changes:_
Added a solution provided by @mithunkalan that prevents the auth flow from being executed twice by the same url. This was an issue impacting both react-native and react-native with expo that caused an "invalid_grant" error.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -225,8 +225,12 @@ export default class AuthClass {
 			});
 
 			// **NOTE** - Remove this in a future major release as it is a breaking change
+			const d = [];
```
[the remark this answers is not in the document]
Comment (ericclemmons) [reply packages/auth/src/Auth.ts]: @iartemiev is working on a test to add in.  I believe he addresses this as well.

@hilsny you've got a lot of 👀 on this one! 🤗

[the remark this answers is not in the document]
Comment (HilSny) [reply packages/auth/src/Auth.ts]: Yeah, thats a really good point @Ashish-Nanda. Let me know if I need to add it or if @iartemiev will handle it.

[the remark this answers is not in the document]
Comment (iartemiev) [reply packages/auth/src/Auth.ts]: @HilSny I can add that in! Will be ready shortly

**Comment by iartemiev** [comment]
This should also address:
#4003
#3550

And possibly others

File: packages/auth/src/Auth.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (iartemiev) [reply packages/auth/src/Auth.ts]: Was running into too many issues when trying to write a unit test for this (RN-specific issues in Jest). We'll either add one later or will test this in integration testing.

We'll get this merged once there's one more approval 👍🏻

**Pull request merged**